drivers/serial/Kconfig.xmc4xxx

```diff
@@ -8,5 +8,6 @@ config UART_XMC4XXX
 	default y
 	depends on DT_HAS_INFINEON_XMC4XXX_UART_ENABLED
 	select SERIAL_HAS_DRIVER
+	select GPIO
```

Comment:
why is this needed? I can't see driver changes using the GPIO API

Reply:
yep, it's no longer needed after I removed gpio dependency, thanks. (same for the other points in your review).